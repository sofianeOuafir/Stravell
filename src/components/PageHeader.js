import React from "react";

const PageHeader = ({ className, children, title }) => (
  <div className={className ? className : "page-header"}>
    <div className="content-container">
      {children ? children : <h1 className="page-header__title">{title}</h1>}
    </div>
  </div>
);

export default PageHeader;
