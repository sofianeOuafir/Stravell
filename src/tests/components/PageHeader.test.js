import React from 'react';
import { shallow } from 'enzyme';
import PageHeader from './../../components/PageHeader';

describe('the component is rendered with a title prop', () => {
  test('should render correctly', () => {
    const props = { title: 'My amazing title' };
    const wrapper = shallow(<PageHeader { ...props } />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('the component is rendered with a prop withSocialShareButtons = false', () => {
  test('should render correctly', () => {
    const props = { title: 'My amazing title', withSocialShareButtons: false };
    const wrapper = shallow(<PageHeader { ...props } />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('the component is rendering the social share buttons by default', () => {
  test('should render correctly', () => {
    const props = { title: 'My amazing title' };
    const wrapper = shallow(<PageHeader { ...props } />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('the component is rendered with a children prop', () => {
  test('should render correctly', () => {
    const component = (
      <PageHeader>
        <p>My amazing children</p>
      </PageHeader>
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});