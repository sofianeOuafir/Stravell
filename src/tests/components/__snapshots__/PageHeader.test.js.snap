// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component is rendered with a children prop should render correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <p>
      My amazing children
    </p>
  </div>
</div>
`;

exports[`the component is rendered with a title prop should render correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      My amazing title
    </h1>
  </div>
</div>
`;
